core/src/test/scala/org/apache/spark/SparkContextSuite.scala

```diff
@@ -233,6 +233,42 @@ class SparkContextSuite extends SparkFunSuite with LocalSparkContext with Eventu
     }
   }
 
+  test("SPARK-30126: addFile when file path contains spaces with recursive works") {
+    withTempDir { dir =>
+      try {
+        val sep = File.separator
+        val tmpDir = Utils.createTempDir(dir.getAbsolutePath + sep + "test space")
+        val tmpConfFile1 = File.createTempFile("test", ".conf", tmpDir)
```

Comment:
can we also add space in the file name to prove it works?

Reply:
ok